jvector-base/src/main/java/io/github/jbellis/jvector/vector/VectorUtilSupport.java

```diff
@@ -113,6 +114,22 @@ public interface VectorUtilSupport {
    */
   float assembleAndSum(VectorFloat<?> data, int baseIndex, ByteSequence<?> baseOffsets, int baseOffsetsOffset, int baseOffsetsLength);
 
+  /**
+   * Calculates the sum of 2 vectors quantized using ProductQuantization.
```

Comment:
Should the documentation say something like "Calculates the dot product between 2 vectors, which were quantized using Product Quantization, using a precomputed table of partial results."? If this is right, should the method name (and its counterparts in other places) better reflect its meaning? Something like "assemblePQDotProduct"?